lib/note-list/no-notes.tsx

```diff
@@ -18,8 +18,12 @@ type EmptyNoteListPlaceholder = {
 const NoNotes = () => {
   const hasLoaded = useSelector((state: S.State) => state.ui.hasLoadedNotes);
   const searchQuery = useSelector((state: S.State) => state.ui.searchQuery);
-  const showTrash = useSelector((state: S.State) => state.ui.showTrash);
-  const openedTag = useSelector((state: S.State) => state.ui.openedTag);
+  const showTrash = useSelector(
+    (state: S.State) => state.ui.showCollection.type === 'trash'
+  );
+  const openedTag = useSelector(
+    (state: S.State) => state.ui.showCollection.tagHash
+  );
```

Comment:
these update together and I don't see any reason to create two hooks for them. if we remodel the data (as noted in the PR comment) then we can get away with replacing the two with a single one.

```ts
const collection = useSelector((state: S.State) => state.ui.collection);

switch (collection.type) {
	case 'tag':
		return {message: `No notes tagged "${collection.tagName}"`, icon: <TagIcon />, button: null};

	case 'trash':
		return {message: 'You trash is empty', icon: <TrashIcon />, button: null};

	case …
}
```

Reply:
I think passing in the one filter makes sense! I honestly didn't even think of it after seeing how many changes I already had, lol. 

What do you think about `collectionFilter`?

Reply:
let's see if @codebykat has thoughts on the name. I'm not as excited about that because it doesn't feel like a "filter" to me.

there's one aspect I haven't commented on, which might be "the elephant in the room" but I haven't raised because it implies somewhat more extensive refactoring, and that's the search query. as it stands it seems proper to keep the search query separate since it applies in all opened collections/views; there's a chance it belongs inside this state description. let's leave that for now though since it might still be best as-is.

that being said the search query is really what "filters" the notes and these are more like drawers we are opening. it's true that they filter, but they represent different spaces or collections.

sorry this must sound pedantic, naming stuff is hard.

Reply:
Okay! I'm honestly not too bothered by the name, renamed to `collection`.

I also combined the different selectors into the file.

Reply:
I'm fine with `collection`.